src/controllers/profile.controller.ts

```diff
@@ -0,0 +1,74 @@
+import { UserService } from "../services/user.services";
+import { Request, Response } from "express";
+
+export const getProfile = async (req: Request, res: Response) => {
+  try {
+    const userId = req.user.id; // Assuming user ID is available in the request object
+    const user = await UserService.getUserByid(userId);
+
+    if (!user) {
+      return res.status(404).json({
+        status: "fail",
+        message: "User not found",
+      });
+    }
+
+    const userWithoutPasswordId = { ...user.dataValues };
+    delete userWithoutPasswordId.password;
+    delete userWithoutPasswordId.id;
+
+    return res.status(200).json({
+      status: "success",
+      data: userWithoutPasswordId,
+    });
+  } catch (error) {
+    console.error("Error fetching profile:", error);
+    return res.status(500).json({
+      status: "error",
+      message: "An error occurred while fetching the profile",
+    });
+  }
+};
+
+
+export const updateProfile = async (req: Request, res: Response) => {
+    try {
+      const userId = req.user.id; // Assuming user ID is available in the request object
```

Comment:
Great job so far but you didn't work on updating profile picture

Reply:
The updating profile picture has been added